src/main/java/at/hannibal2/skyhanni/features/mining/glacitemineshaft/CorpseTracker.kt

```diff
@@ -0,0 +1,142 @@
+import at.hannibal2.skyhanni.utils.LorenzUtils.isInIsland
+import at.hannibal2.skyhanni.utils.NEUInternalName
+import at.hannibal2.skyhanni.utils.NumberUtil.addSeparators
+import at.hannibal2.skyhanni.utils.NumberUtil.shortFormat
+import at.hannibal2.skyhanni.utils.renderables.Renderable
+import at.hannibal2.skyhanni.utils.renderables.Searchable
+import at.hannibal2.skyhanni.utils.renderables.toSearchable
+import at.hannibal2.skyhanni.utils.tracker.BucketedItemTrackerData
+import at.hannibal2.skyhanni.utils.tracker.ItemTrackerData.TrackedItem
+import at.hannibal2.skyhanni.utils.tracker.SkyHanniBucketedItemTracker
+import com.google.gson.annotations.Expose
+import net.minecraftforge.fml.common.eventhandler.SubscribeEvent
+import java.util.EnumMap
+
+@SkyHanniModule
+object CorpseTracker {
+    private val config get() = SkyHanniMod.feature.mining.glaciteMineshaft.corpseTracker
+
+    private val tracker = SkyHanniBucketedItemTracker<CorpseType, BucketData>(
+        "Corpse Tracker",
+        { BucketData() },
+        { it.mining.mineshaft.corpseProfitTracker },
+    ) { drawDisplay(it) }
+
+    class BucketData : BucketedItemTrackerData<CorpseType>() {
+        override fun resetItems() {
+            corpsesLooted = EnumMap(CorpseType::class.java)
+        }
+
+        override fun getDescription(timesGained: Long): List<String> {
+            val divisor = 1.coerceAtLeast(getSelectedBucket()?.let { corpsesLooted[it]?.toInt() } ?: corpsesLooted.sumAllValues().toInt())
+            val percentage = timesGained.toDouble() / divisor
+            val dropRate = LorenzUtils.formatPercentage(percentage.coerceAtMost(1.0))
+            return listOf(
+                "§7Dropped §e${timesGained.addSeparators()} §7times.",
+                "§7Your drop rate: §c$dropRate.",
+            )
+        }
+
+        override fun getCoinName(item: TrackedItem) = "<no coins>"
+        override fun getCoinDescription(item: TrackedItem): List<String> = listOf("<no coins>")
+
+        @Expose
+        var corpsesLooted: MutableMap<CorpseType, Long> = EnumMap(CorpseType::class.java)
+
+        fun getCorpseCount(): Long = getSelectedBucket()?.let { corpsesLooted[it] } ?: corpsesLooted.values.sum()
+    }
+
+    private fun addLootedCorpse(type: CorpseType) = tracker.modify { it.corpsesLooted.addOrPut(type, 1) }
+
+    @SubscribeEvent
+    fun onCorpseLoot(event: CorpseLootedEvent) {
+        addLootedCorpse(event.corpseType)
+        for ((itemName, amount) in event.loot) {
+            NEUInternalName.fromItemNameOrNull(itemName)?.let { item ->
+                tracker.modify {
+                    it.addItem(event.corpseType, item, amount)
+                }
+            }
+        }
+    }
+
+    private fun drawDisplay(bucketData: BucketData): List<Searchable> = buildList {
+        addSearchString("§b§lGlacite Corpse Profit Tracker")
+        tracker.addBucketSelectorsReified<CorpseType>(this, bucketData, "Corpse Type")
```

Comment:
All `Tracker...`s' `drawDisplay` methods take their `data: Data` object as an input - don't know why this would be different. The bucketed data does exist in the tracker, but not in an accessible manner - `drawDisplay(it)` is common across all others, don't think this needs changing.